Core/GDCore/Extensions/Metadata/BehaviorMetadata.cpp

```diff
@@ -42,6 +43,27 @@ BehaviorMetadata::BehaviorMetadata(
   if (sharedDatasInstance) sharedDatasInstance->SetTypeName(name_);
 }
 
+BehaviorMetadata::BehaviorMetadata(
+    const gd::String& extensionNamespace,
+    const gd::String& name_,
+    const gd::String& fullname_,
+    const gd::String& description_,
+    const gd::String& group_,
+    const gd::String& icon24x24_): BehaviorMetadata(
+        extensionNamespace,
+        name_,
+        fullname_,
+        // Default name is the name
+        name_,
+        description_,
+        group_,
+        icon24x24_,
+        // Class name is the name, actually unused
+        name_,
+        // It is only used to get the name for GetName.
+        gd::make_unique<gd::Behavior>(),

```

Comment:
It's not ideal as `Get()` will still returns something. We could maybe use a boolean member to return null. This way the class will have a clean interface.

Reply:
Having a nullptr (which is what a Unique pointer to nothing is) is fine I guess for now?

Reply:
No, I forgot that it doesn't call the Behavior default constructor.